ProjFS.Mac/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -664,14 +665,32 @@ bool VirtualizationRoots_ProcessMayAccessOfflineRoots(pid_t pid)
     
     RWLock_AcquireShared(s_virtualizationRootsLock);
     {
-        for (uint32_t i = 0; i < s_offlineIOPIDCount; ++i)
+        while (true)
         {
-            if (s_offlineIOPIDs[i] == pid)
+            for (uint32_t i = 0; i < s_offlineIOPIDCount; ++i)
+            {
+                if (s_offlineIOPIDs[i] == pid)
+                {
+                    result = true;
+                    goto done;
+                }
+            }
+
+            proc_t process = proc_find(pid);
+            if (process == nullptr)
+            {
+                break;
+            }
+            
+            pid = proc_ppid(process);
+            proc_rele(process);
+            
+            if (pid <= 1)
```

Comment:
Is the idea we walk all the parents to check for the pid allowed to register.  Let's please add some comments here so it's super obvious whats going on.